ibctest/relayer.go

```diff
@@ -197,12 +220,24 @@ func (r *Relayer) StartRelayer(ctx context.Context, _ ibc.RelayerExecReporter, p
 	r.errCh = make(chan error, 1)
 	ctx, r.cancel = context.WithCancel(ctx)
 
-	args := append([]string{"--processor=events"}, pathNames...)
+	var args []string
+	if r.config.Processor == "" {
+		r.config.Processor = relayer.ProcessorEvents
+	}
+	args = append(args,
+		"--processor", r.config.Processor,
+		"--block-history", strconv.FormatUint(r.config.InitialBlockHistory, 10),
+	)
+
+	args = append(args, pathNames...)
 	go r.start(ctx, args...)
 	return nil
 }
 
 func (r *Relayer) StopRelayer(ctx context.Context, _ ibc.RelayerExecReporter) error {
+	if r.cancel == nil {
+		return nil
+	}
```

Comment:
Why guard against a nil `cancel` here?

The doc comment on the cancel field says it is set in `StartRelayer`. It doesn't seem valid to call `StopRelayer` without first calling `StartRelayer`.

Reply:
ibctest makes assumptions that StopRelayer is idempotent, e.g.:
https://github.com/strangelove-ventures/ibctest/blob/7653dfa748149c0bc92277e752b35bd4c2f7b94e/test_setup.go#L101

Reply:
Okay. I didn't expect that.